app/models/post.rb

```diff
@@ -0,0 +1,23 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_create :update_user_posts_counter
+
+  def update_user_posts_counter
+    user.increment!(:posts_counter)
+  end
+
+  def recent_comments(limit = 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
+
+  def update_comments_counter
+    update(comments_counter: comments.count)
+  end
+
+  def update_likes_counter
+    update(likes_counter: likes.count)
+  end
```

Comment:
- [ ] as describre in the requirements , the `Post` model should have two methods 👆  : A method that updates the posts counter for a user, and a  method which returns the 5 most recent comments for a given post. 💯 . Kindly make sure you have these two method 👍. For the othe two method `update_comments_counter` and ` update_likes_counter` , you have already created them respectively in the `Comment` and `Like` model 😄 it is not necessary to put them here 👍 

![Capture](https://github.com/uno36/blog-app-rails/assets/80458199/0fe8b9bd-4974-4775-bf34-7e60710f0c6e)

- [ ] Kindly use `after_save` because as for a database that can be updated , I think `after_save` it prefered because `after_create` has no effect on `update` 😄 

- [ ] Only the **methods** for updating should be private because they are excecuted internally 😄 , Kindly update that 👍 

- [ ] [OPTIONAL] I will suggest to use `increment` instead of `update` because `increment` method performs the increment operation directly on the database level, ensuring atomicity. it will increase the efficiency as you are perfomiing simple logic 👍 . Kindly check[ here](https://stackoverflow.com/questions/47790927/rails-increment-a-column-and-update-attributes) for more details